### PR TITLE
Add actual Plots tests

### PR DESCRIPTION
**Pull request by briochemc**
**This is not for merging (yet)!** I noticed that `plot(x,y,z)` threw an error when using Unitful arrays, so I added some tests here to show them to you. (BTW I'm more than hapy to help and I tried to fix it but I'm not even sure how the `@recipe` macro works, so I did not go very far... so kind'of counting on you here 😅)

### CODE REVIEW

**Comment by briochemc** [comment]
FYI here are the tests results, which show all the failures:
```julia
Test Summary:                   | Pass  Error  Total
Plots                           |   19     20     39
  One array                     |    5             5
  Two arrays                    |    9             9
  Three arrays                  |          13     13
  Unitful/unitless combinations |    5      7     12
    plot(A, A)                  |    1             1
    plot(A, Q)                  |    1             1
    plot(Q, A)                  |    1             1
    plot(Q, Q)                  |    1             1
    plot(A, A, A)               |    1             1
    plot(A, A, Q)               |           1      1
    plot(A, Q, A)               |           1      1
    plot(A, Q, Q)               |           1      1
    plot(Q, A, A)               |           1      1
    plot(Q, A, Q)               |           1      1
    plot(Q, Q, A)               |           1      1
    plot(Q, Q, Q)               |           1      1
```

**Comment by briochemc** [comment]
I think this fixes the bug.

AFAIU, for 3 variables (like in `plot(x,y,z)`) there are already defined recipes with signatures like, e.g., 

```julia
apply_recipe(..., x::AbstractArray{T,1} where T, y::AbstractArray{T,1} where T, z::AbstractArray{T,1} where T)
```

which are more specific than what we had before (although the reason why remains elusive to me). This meant that `apply_recipe` did **not** dispatch on our recipes such as 

```julia
apply_recipe(..., x::Q, y::Q, z::Q)
```

The solution in this last commit is to define more specific `apply_recipe` methods, e.g.,

```julia
apply_recipe(..., x::AbstractArray{T,1} where T<:Quantity, y::AbstractArray{T,1} where T, z::AbstractArray{T,1} where T)
```

---

Note that in order to not add many lines, I used a loop with an `@eval` for all the 3-variable recipes. I hope that's OK 😅 

Let me know what you think!

**Comment by briochemc** [comment]
(Sorry did not mean to close it!)

**Comment by briochemc** [comment]
Also for discussion, I just read https://docs.julialang.org/en/v1/manual/methods/#man-methods-orthogonalize-1
and realized that maybe someone might suggest a better approach to these recipes :)

File: src/UnitfulRecipes.jl [hunk 1]
```diff
@@ -28,12 +28,7 @@ Mutates `attr` by converting/removing unitful attributes.
 function resolve_axis!(attr, arr::AbstractArray{T}, axis::Int) where {T<:Quantity}
     # convert (if user-provided unit) and strip unit from data
     key = key_unit(axis)
-    if haskey(attr, key)
-        u = attr[key] # 
-        delete!(attr, key)
-    else
-        u = unit(T)
-    end
+    u = pop!(attr, key, unit(T))
```
Comment (jw3126) [inline src/UnitfulRecipes.jl:31]: Cool, somehow I never came across that `pop!` method! 👍

**Review by jw3126 — APPROVED** [review]
Thanks a lot for fixing this! Looks good! As for a more orthogonal design, I am skeptic. The actual implementation in `recipe!` is quite clean and orthogonal I would say. But for hooking it up with `Plots.jl` I suspect to define less recipes, would require unrealistic changes to `Plots.jl`.

**Comment by jw3126** [comment]
BTW @briochemc feel free to merge and tag a release!

**Comment by jw3126** [comment]
Also thanks a lot for the analysis and finding the interesting dispatch behaviour!

**Pull request merged**